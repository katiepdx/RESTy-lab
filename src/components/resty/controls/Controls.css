.userForm {
  background-color: whitesmoke;
  padding: 2rem;
  font-size: 1.2rem;
  line-height: 2.3rem;
  
  display: flex;
  flex-direction: column;;
  justify-content: space-evenly;

  border-radius: 2rem;
}

button {
  width: fit-content;
  padding: 0 2rem;
  border-radius: 6px;
  background-color: rgba(245, 241, 241, 0.726);
  border: solid 1px;
}
