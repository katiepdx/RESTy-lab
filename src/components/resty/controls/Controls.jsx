// url search bar - input field for URL
// radio buttons GET POST PUT DELETE
// input field for typing JSON for PUT POST
import React from 'react';
import propTypes from 'prop-types';
import styles from './Controls.css'

const Controls = ({ url, json, onChange, onClick }) => (
  <section className={styles.userForm}>
    <div>
      <label htmlFor="url">Search Bar</label>
      <input data-testid="url-input" type="text" name="url" value={url} onChange={onChange} />
    </div>
    <div>
      <label htmlFor="method">Method</label>
      <input data-testid="get-input" type="radio" name="method" value="GET" onChange={onChange} /> GET
      <input data-testid="post-input" type="radio" name="method" value="POST" onChange={onChange} /> POST
      <input data-testid="put-input" type="radio" name="method" value="PUT" onChange={onChange} /> PUT
      <input data-testid="delete-input" type="radio" name="method" value="DELETE" onChange={onChange} /> DELETE
    </div>
    <div>
      <label htmlFor="json">JSON (POST/PUT)</label>
      <textarea data-testid="json-input" type="text" name="json" value={json} onChange={onChange} />
    </div>
    <button data-testid="submit-button" onClick={onClick}>Go!</button>
  </section>
);

Controls.propType = {
  url: propTypes.string.isRequired,
  json: propTypes.string,
  onChange: propTypes.func.isRequired,
  onClick: propTypes.func.isRequired
};

export default Controls;
