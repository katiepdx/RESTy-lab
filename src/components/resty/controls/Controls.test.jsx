
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());

  it('renders Controls', () => {
    const { asFragment } = render(<Controls
      url="www.api1.com"
      method="POST"
      json="test json"
      onChange={() => {}}
      onClick={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
