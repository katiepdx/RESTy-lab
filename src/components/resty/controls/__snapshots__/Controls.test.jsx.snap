// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<DocumentFragment>
  <section
    class="userForm"
  >
    <div>
      <label
        for="url"
      >
        Search Bar
      </label>
      <input
        data-testid="url-input"
        name="url"
        type="text"
        value="www.api1.com"
      />
    </div>
    <div>
      <label
        for="method"
      >
        Method
      </label>
      <input
        data-testid="get-input"
        name="method"
        type="radio"
        value="GET"
      />
       GET
      <input
        data-testid="post-input"
        name="method"
        type="radio"
        value="POST"
      />
       POST
      <input
        data-testid="put-input"
        name="method"
        type="radio"
        value="PUT"
      />
       PUT
      <input
        data-testid="delete-input"
        name="method"
        type="radio"
        value="DELETE"
      />
       DELETE
    </div>
    <div>
      <label
        for="json"
      >
        JSON (POST/PUT)
      </label>
      <textarea
        data-testid="json-input"
        name="json"
        type="text"
      >
        test json
      </textarea>
    </div>
    <button
      data-testid="submit-button"
    >
      Go!
    </button>
  </section>
</DocumentFragment>
`;
